lib

```diff
@@ -16,6 +16,31 @@ function current-tty {
     fi
 }
 
+# Find the top level parent PID of current shell (not including root process), also accounting for TMUX
+function top-level-ppid {
```

Comment:
This function will infinitely recurse if `ps` doesn’t take the expected options (maybe on other BSDs? I dunno), or if there is no process with PID 1 (maybe in a container?).

I’m not especially worried about this, but it could be prevented by setting `setopt localoptions errexit` here.

Reply:
> but it could be prevented by setting `setopt localoptions errexit` here

Tried to google this option and barely found anything. I think I fixed it much simpler :)

Reply:
That works too!

`setopt localoptions errexit` might be hard to google because zsh ignores underscores in the options. Just running `setopt` gives a list of options, which are reported without underscores, but the [manual](https://zsh.sourceforge.io/Doc/Release/index.html#Top) lists them with underscores. So the above is equivalent to `setopt local_options err_exit`.

* `local_options` causes the shell options to only be set for the duration of the local function.
* `err_exit` causes the shell to exit when a command fails, like `set -e`. (It should probably have been `err_return` which causes the function to return.)

Docs:
* `setopt` in https://zsh.sourceforge.io/Doc/Release/Shell-Builtin-Commands.html for
* `LOCAL_OPTIONS` and `ERR_EXIT` in https://zsh.sourceforge.io/Doc/Release/Options.html (but `ERR_RETURN` might have been better)